libraries/libfc/include/fc/crypto/bls_public_key.hpp

```diff
@@ -26,54 +21,30 @@ namespace fc { namespace crypto { namespace blslib {
          bls_public_key() = default;
          bls_public_key( bls_public_key&& ) = default;
          bls_public_key( const bls_public_key& ) = default;
-         bls_public_key& operator= (const bls_public_key& ) = default;
-
-         bls_public_key( bls12_381::g1 pkey ){
+         bls_public_key( const bls12_381::g1& pkey ){

```

Comment:
Please make this explicit and one line, as in:
`explicit bls_public_key( const bls12_381::g1& pkey ) : _pkey(pkey) {}`

remove unnecessary `friend` declarations. Remove extra spaces.

Class can be simplified to:

```
   struct bls_public_key
   {
         bls_public_key() = default;
         bls_public_key(bls_public_key&&) = default;
         bls_public_key(const bls_public_key&) = default;
         explicit bls_public_key(const bls12_381::g1& pkey) : _pkey(pkey) {}
         explicit bls_public_key(const std::string& base58str);

         bls_public_key& operator= (const bls_public_key&) = default;
         std::string to_string(const yield_function_t& yield = yield_function_t()) const;
         friend bool operator==( const bls_public_key& p1, const bls_public_key& p2);

         bls12_381::g1 _pkey;
   }; 
```